internal/golang/elf.go

```diff
@@ -1,4 +1,4 @@
-//go:build go1.21
+//go:build go1.23
```

Comment:
This is not necessary to change Go version here. Whatever's in the file works since go1.21, and go1.23 sets all the previous versions tags.